### PR TITLE
Fix spring properties

### PR DESCRIPTION
**Pull request by v1r3n**

### CODE REVIEW

File: src/main/java/io/orkes/conductor/client/ApiClient.java [hunk 1]
```diff
@@ -111,6 +111,9 @@ public ApiClient(String basePath, SecretsManager secretsManager, String keyPath,
     }
 
     public ApiClient(String basePath, String keyId, String keySecret) {
+        if(basePath == null) {
+            basePath = "http://localhost:8080/api";
```
Comment (coderabhigupta) [inline src/main/java/io/orkes/conductor/client/ApiClient.java:115]: So default to localhost?

File: src/main/java/io/orkes/conductor/client/spring/OrkesSpringWorkerConfiguration.java [hunk 2]
```diff
@@ -28,19 +28,26 @@ public OrkesSpringWorkerConfiguration(Environment environment) {
...
     public String getDomain(String taskName) {
-        String key = "conductor.worker." + taskName + ".domain";
-        return environment.getProperty(key, String.class, null);
+        return getProperty(taskName, "domain", String.class, null);
+    }
+
+    private <T>T getProperty(String taskName, String property, Class<T> type, T defaultValue) {
+        String key = "conductor.worker." + taskName + "." + property;
+        T value = environment.getProperty(key, type, defaultValue);
+        if(value == null || value == defaultValue) {
```
Comment (coderabhigupta) [inline src/main/java/io/orkes/conductor/client/spring/OrkesSpringWorkerConfiguration.java:47]: Can the `value` ever be null, now that you have a defaultValue?

File: src/test/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,7 @@
+conductor.worker.all.domain=domainx
+conductor.worker.all.pollingInterval=3
+#conductor.worker.all.threadCount=7
+
+conductor.worker.hello.domain=helo_domain
+conductor.worker.hello.pollingInterval=1
+conductor.worker.hello.threadCount=13
```
Comment (coderabhigupta) [inline src/test/resources/application.properties:7]: Do we want to retain these?

File: src/main/java/io/orkes/conductor/client/ApiClient.java (hunk 1 above)
Comment (v1r3n) [reply src/main/java/io/orkes/conductor/client/ApiClient.java:115]: yes.

Comment (v1r3n) [reply src/main/java/io/orkes/conductor/client/ApiClient.java:115]: this has been there always when you init without any args.

File: src/main/java/io/orkes/conductor/client/spring/OrkesSpringWorkerConfiguration.java (hunk 2 above)
Comment (v1r3n) [reply src/main/java/io/orkes/conductor/client/spring/OrkesSpringWorkerConfiguration.java:47]: value can be null if not set.

**Review by coderabhigupta — APPROVED** [review]

**Pull request merged**